pkg/cachemanager/cachemanager_test/cachemanager_integration_test.go

```diff
@@ -0,0 +1,346 @@
+	require.NoError(t, cacheManager.UpsertSource(ctx, syncSourceOne, []schema.GroupVersionKind{configMapGVK}))
+
+	expected = map[cachemanager.CfDataKey]interface{}{
+		{Gvk: configMapGVK, Key: "default/config-test-1"}: nil,
+		{Gvk: configMapGVK, Key: "default/config-test-2"}: nil,
+	}
+	require.Eventually(t, expectedCheck(cfClient, expected), eventuallyTimeout, eventuallyTicker)
+	// now assert that the gvkAggregator looks as expected
+	agg.IsPresent(configMapGVK)
+	gvks = agg.List(syncSourceOne)
+	require.Len(t, gvks, 1)
+	_, foundConfigMap = gvks[configMapGVK]
+	require.True(t, foundConfigMap)
+	_, foundPod = gvks[podGVK]
+	require.False(t, foundPod)
+
+	// now make sure that adding another sync source with the same gvk has no side effects
+	syncSourceTwo := aggregator.Key{Source: "source_b", ID: "ID_b"}
+	require.NoError(t, cacheManager.UpsertSource(ctx, syncSourceTwo, []schema.GroupVersionKind{configMapGVK}))
+	agg.IsPresent(configMapGVK)
+	gvks = agg.List(syncSourceTwo)
+	require.Len(t, gvks, 1)
+	_, foundConfigMap = gvks[configMapGVK]
+	require.True(t, foundConfigMap)
+
+	require.NoError(t, cacheManager.UpsertSource(ctx, syncSourceOne, []schema.GroupVersionKind{podGVK}))
+	expected2 := map[cachemanager.CfDataKey]interface{}{
+		{Gvk: configMapGVK, Key: "default/config-test-1"}: nil,
+		{Gvk: configMapGVK, Key: "default/config-test-2"}: nil,
+		{Gvk: podGVK, Key: "default/pod-1"}:               nil,
+	}
+	require.Eventually(t, expectedCheck(cfClient, expected2), eventuallyTimeout, eventuallyTicker)
+
+	// now go on and unreference sourceTwo's gvks; this should schedule the config maps to be removed
+	require.NoError(t, cacheManager.UpsertSource(ctx, syncSourceTwo, []schema.GroupVersionKind{}))
+	expected3 := map[cachemanager.CfDataKey]interface{}{
+		// config maps no longer required by any sync source
+		// {Gvk: configMapGVK, Key: "default/config-test-1"}: nil,
+		// {Gvk: configMapGVK, Key: "default/config-test-2"}: nil,
+		{Gvk: podGVK, Key: "default/pod-1"}: nil,
+	}
+	require.Eventually(t, expectedCheck(cfClient, expected3), eventuallyTimeout, eventuallyTicker)
+
+	// now remove all the sources
+	require.NoError(t, cacheManager.RemoveSource(ctx, syncSourceTwo))
+	require.NoError(t, cacheManager.RemoveSource(ctx, syncSourceOne))
+
+	// and expect an empty cache and empty aggregator
+	require.Eventually(t, expectedCheck(cfClient, map[cachemanager.CfDataKey]interface{}{}), eventuallyTimeout, eventuallyTicker)
+	require.True(t, len(agg.GVKs()) == 0)
+
+	// cleanup
+	require.NoError(t, c.Delete(ctx, cm), "deleting ConfigMap config-test-1")
+	require.NoError(t, c.Delete(ctx, cm2), "deleting ConfigMap config-test-2")
+	require.NoError(t, c.Delete(ctx, pod), "deleting Pod pod-1")
+}
+
+// TestCacheManager_ExcludeProcesses makes sure that changing the process excluder
+// in the cache manager triggers a re-evaluation of GVKs.
+func TestCacheManager_ExcludeProcesses(t *testing.T) {
```

Comment:
I believe we could get more value by combining an exhaustive unit test of ExcludeProcesses with an e2e test.

The uni test would allow us to better confirm that the CacheManager is changed correctly by the ExcludeProcesses function.

We'll want the e2e test anyways since it shows us that the product truly works.